### PR TITLE
feat: #51 show goal and goal metrics near way table

### PR DESCRIPTION
**Pull request by dayanych**
### Overview

Added the ability to edit a goal and show in `WayPage`

### Changes description
- `updateGoalDTO` and `updateGoalPreview` were realized
- `EditableTextarea` was added
- `renderCellValue` was deleted from mw-webapp/src/logic/waysTable/columns.tsx

### CODE REVIEW

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 1]
```diff
@@ -59,6 +61,17 @@ export const WayPage = (props: WayPageProps) => {
             onChangeFinish={(text) => changeWayName(way, text)}
             isEditable={true}
           />
+          <EditableTextarea
+            text={way.goal.description}
+            onChangeFinish={(description) => {
+              const goalDTO = {
+                ...way.goal,
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:68]: We don't use DTO in UI layer. Looks like it's wrong naming in your case (goalDTO => goal). Please check does you understand models right) If you  have questions call me in Telegram

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 2]
```diff
@@ -59,6 +61,17 @@ export const WayPage = (props: WayPageProps) => {
             onChangeFinish={(text) => changeWayName(way, text)}
             isEditable={true}
           />
+          <EditableTextarea
+            text={way.goal.description}
+            onChangeFinish={(description) => {
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:66]: Please move out this function from JSX

File: mw-webapp/src/logic/waysTable/columns.module.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
+.goalDescription {
+  display: -webkit-box;
+  overflow: hidden;
+  max-width: 150px;
+  -webkit-box-orient: vertical;
+  -webkit-line-clamp: 2;
+  text-overflow: ellipsis;
+  word-wrap: break-word;
+}
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/waysTable/columns.module.scss:9]: Why do we use webkit styles :
https://stackoverflow.com/questions/74603094/what-does-display-webkit-box-do-and-what-browsers-support-it

File: mw-webapp/src/logic/wayPage/WayPage.tsx (hunk 1 above)
Comment (dayanych) [reply mw-webapp/src/logic/wayPage/WayPage.tsx:68]: Done.

File: mw-webapp/src/component/editableText/renderSpan.tsx [hunk 4]
```diff
@@ -24,7 +24,7 @@ const renderEmptySpan = () => (
  * TODO: move to separate component, task #208
  */
 export const renderSpan = (value: string | number, isDone?: boolean) => (
-  (value === UnicodeSymbols.ZERO_WIDTH_SPACE)
+  (value === UnicodeSymbols.ZERO_WIDTH_SPACE || !value)
     ? renderEmptySpan()
```
[the remark this answers is not in the document]
Comment (dayanych) [reply mw-webapp/src/component/editableText/renderSpan.tsx:28]: Removed ZERO_WIDTH_SPACE in the condition.

File: mw-webapp/src/component/editableText/renderSpan.tsx [hunk 5]
```diff
@@ -13,18 +12,18 @@ const renderSpanWithValue = (value: string | number, isDone?: boolean) => (
 /**
  * Render empty span
  */
-const renderEmptySpan = () => (
+const renderEmptySpan = (placeholderSpanText: string | undefined) => (
```
Comment (tritonJS826) [inline mw-webapp/src/component/editableText/renderSpan.tsx:15]:
```suggestion
const renderEmptySpan = (placeholderSpanText?: string) => (
```

File: mw-webapp/src/logic/waysTable/columns.tsx [hunk 6]
```diff
@@ -68,7 +78,11 @@ export const columns = [
      */
     cell: ({row}) => {
       return (
-        row.original.currentMentors.map((currentMentor) => renderCellValue(currentMentor.name))
+        row.original.currentMentors.map((currentMentor) => (
+          <div key={currentMentor.uuid}>
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/columns.tsx:82]: if you are using div element only for adding key attribute you can use React.Fragment

**Review by tritonJS826 — APPROVED** [review]

**Review by Ekaterina1994 — APPROVED** [review]

**Review by Howl404 — APPROVED** [review]

File: mw-webapp/src/logic/waysTable/columns.tsx (hunk 6 above)
Comment (dayanych) [reply mw-webapp/src/logic/waysTable/columns.tsx]: Done.

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**